### PR TITLE
fix(4687): rest status codes

### LINKED ISSUE #4687: [BUG] Improper status codes for snapshot executions
**Describe the bug**
A missing S3 bucket for the snapshot results on a 503

```
{
    "error": {
        "root_cause": [
            {
                "type": "repository_exception",
                "reason": "[snapshot-name] could not read repository data from index blob"
            }
        ],
        "type": "repository_exception",
        "reason": "[snapshot-name] could not read repository data from index blob",
        "caused_by": {
            "type": "i_o_exception",
            "reason": "Exception when listing blobs by prefix [index-name]",
            "caused_by": {
                "type": "amazon_s3_exception",
                "reason": "amazon_s3_exception: The specified bucket does not exist (Service: Amazon S3; Status Code: 404; Error Code: NoSuchBucket; Request ID: V3DXMMJXXXXXX; S3 Extended Request ID: D7W3ThAkomdyqtKj0msy1g1Q/qpU0h9XXXXXXXX)"
            }
        }
    },
    "status": 500
}
```

Then a concurrent snapshot exception is a 503 

https://github.com/opensearch-project/OpenSearch/blob/15161a86491569d94642f3a88174e1389218b303/server/src/main/java/org/opensearch/snapshots/ConcurrentSnapshotExecutionException.java#L45-L62

**To Reproduce**

**Expected behavior**
Errors which are client side misconfiguration or client side operations which aren't supported based on the system's concurrency specification should be a 4xx

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by p1729**
### Description
[Describe what this change achieves]

### Issues Resolved
#4687 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by p1729** [comment]
@Bukhtawar I have opened this PR to get your early feedback on the expected solution. I will be adding tests as well if needed soon.

File: server/src/main/java/org/opensearch/repositories/RepositoryException.java [hunk 1]
```diff
@@ -69,6 +70,15 @@ public RepositoryException(StreamInput in) throws IOException {
         repository = in.readOptionalString();
     }
 
+    @Override
+    public RestStatus status() {
+        String causeDescription = unwrapCause().getMessage();
+        if (causeDescription != null && causeDescription.contains("not found")) {
```
Comment (dblock) [inline server/src/main/java/org/opensearch/repositories/RepositoryException.java:76]: This isn't very reliable. 

Start by writing a unit test that reproduces the problem, then we can talk about how to fix it best.

Comment (p1729) [reply server/src/main/java/org/opensearch/repositories/RepositoryException.java:76]: make sense, I will try to write an integration/sociable test.

**Comment by minalsha** [comment]
@p1729 could you please share status of the progress here for this PR. PLease let us know if you need any help. Thanks

**Comment by dreamer-89** [comment]
@p1729: Are you still working on this PR ?

**Pull request closed without merging**